### PR TITLE
mpOpenApi-2.0 merging fixes

### LINKED ISSUE #19214: mpOpenApi-2.0 should detect when extensions clash at the top level
OpenAPI allows arbitrary extensions to be added to any model object.

If two models to be merged contain an extension with the same name at the top level but with different values, this should be detected as a clash and cause one of the models not to be included in the merged output.

E.g.
```yaml
openapi: 3.0.3
info:
  title: "test1"
  version: "1.0"
x-foo: "bar"
```
```yaml
openapi: 3.0.3
info:
  title: "test2"
  version: "1.0"
x-foo: "baz"
```

Similar to a path name clash, we should output a warning when this occurs.

In the current code, the extension from the second model to be merged will overwrite the extension from the first model, making it essentially arbitrary which extension appears in the merged model.

### LINKED ISSUE #19215: mpOpenApi-2.0: Server elements can be created incorrectly when a merge is attempted but all but one module is excluded
When a merge occurs and all but one document is excluded from the merge due to clashes, the remaining document is returned by `MergeProcessor` in its original form but the application path is not set correctly.

This can result in the generated server elements being incorrect when the OpenAPI document is returned to the user.

### PR DESCRIPTION
**Pull request by Azquelt**
* Detect extension clashes
* Return the correct application path when a merge clash occurs

Fixes #19215 
Fixes #19214

### CODE REVIEW

**Comment by Azquelt** [comment]
#build
#libby

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 1]
```diff
@@ -73,6 +73,10 @@ OPENAPI_MERGE_PROBLEM_PATH_CLASH=The {0} path from the {1} module clashes with a
 OPENAPI_MERGE_PROBLEM_PATH_CLASH.explanation=The path is two modules. Both instances of the path cannot be included in the same OpenAPI document. This message is returned as part of CWWKO1662.
 OPENAPI_MERGE_PROBLEM_PATH_CLASH.useraction=Either change one of the conflicting modules or configure the mpOpenApi-2.0 feature to exclude one of them. This problem can be caused if servers are included in the OpenAPI document and the server URL does not end with the context root of the application (for example, because the application is accessed at a different URL through a reverse proxy).
 
+OPENAPI_MERGE_PROBLEM_EXTENSION_CLASH=The {0} extension key from the {1} module clashes with the same extension key from the {2} module. {1} will not be merged.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops]: per IBM style, avoid future tense

{1} will not be merged
--->
{1} cannot be merged

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 2]
```diff
@@ -73,6 +73,10 @@ OPENAPI_MERGE_PROBLEM_PATH_CLASH=The {0} path from the {1} module clashes with a
...
+OPENAPI_MERGE_PROBLEM_EXTENSION_CLASH.explanation=Two modules declare the same extension with different values at the top level of the OpenAPI document. Both instances of the extension cannot be included in the same OpenAPI document. This message is returned as part of CWWKO1662.
+OPENAPI_MERGE_PROBLEM_EXTENSION_CLASH.useraction=Either change one of the conflicting modules to remove the extension or configure the mpOpenApi-2.0 feature to exclude one of them.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops]: Suggested edit:

Either change one of the conflicting modules to remove the extension or configure the mpOpenApi-2.0 feature to exclude one of the modules.

**Review by dmuelle — CHANGES_REQUESTED** [review]

**Comment by Azquelt** [comment]
Thank you @dmuelle, I have made both changes.

**Comment by Azquelt** [comment]
#build
#libby

**Review by tevans78 — APPROVED** [review]

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 3]
```diff
@@ -69,9 +69,13 @@ OPENAPI_MERGE_PROBLEMS_CWWKO1662W=CWWKO1662W: The following problems occurred wh
...
+OPENAPI_MERGE_PROBLEM_PATH_CLASH=The {0} path from the {1} module clashes with a path from the {2} module. {1} cannot be merged.
 OPENAPI_MERGE_PROBLEM_PATH_CLASH.explanation=The path is two modules. Both instances of the path cannot be included in the same OpenAPI document. This message is returned as part of CWWKO1662.
-OPENAPI_MERGE_PROBLEM_PATH_CLASH.useraction=Either change one of the conflicting modules or configure the mpOpenApi-2.0 feature to exclude one of them. This problem can be caused if servers are included in the OpenAPI document and the server URL does not end with the context root of the application (for example, because the application is accessed at a different URL through a reverse proxy).
+OPENAPI_MERGE_PROBLEM_PATH_CLASH.useraction=Either change one of the conflicting modules or configure the mpOpenApi-2.0 feature to exclude one of the modules. This problem can be caused if servers are included in the OpenAPI document and the server URL does not end with the context root of the application (for example, because the application is accessed at a different URL through a reverse proxy).
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops]: suggested edit:
... This problem might occur if servers are included in the OpenAPI document and the server URL does not end with the context root of the application. For example, when the application is accessed at a different URL through a reverse proxy.

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 4]
```diff
@@ -69,9 +69,13 @@ OPENAPI_MERGE_PROBLEMS_CWWKO1662W=CWWKO1662W: The following problems occurred wh
 OPENAPI_MERGE_PROBLEMS_CWWKO1662W.explanation=The OpenAPI documentation from multiple web modules cannot be successfully merged. The endpoints from the reported modules are not included in the generated OpenAPI document.
 OPENAPI_MERGE_PROBLEMS_CWWKO1662W.useraction=Review the merge problems to identify why the generated OpenAPI documents cannot be merged.
 
-OPENAPI_MERGE_PROBLEM_PATH_CLASH=The {0} path from the {1} module clashes with a path from the {2} module. {1} will not be merged.
+OPENAPI_MERGE_PROBLEM_PATH_CLASH=The {0} path from the {1} module clashes with a path from the {2} module. {1} cannot be merged.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops]: {1} cannot be merged.
-->
The  {1} module cannot be merged.

File: dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops [hunk 5]
```diff
@@ -69,9 +69,13 @@ OPENAPI_MERGE_PROBLEMS_CWWKO1662W=CWWKO1662W: The following problems occurred wh
...
 OPENAPI_MERGE_PROBLEM_PATH_CLASH.explanation=The path is two modules. Both instances of the path cannot be included in the same OpenAPI document. This message is returned as part of CWWKO1662.
-OPENAPI_MERGE_PROBLEM_PATH_CLASH.useraction=Either change one of the conflicting modules or configure the mpOpenApi-2.0 feature to exclude one of them. This problem can be caused if servers are included in the OpenAPI document and the server URL does not end with the context root of the application (for example, because the application is accessed at a different URL through a reverse proxy).
+OPENAPI_MERGE_PROBLEM_PATH_CLASH.useraction=Either change one of the conflicting modules or configure the mpOpenApi-2.0 feature to exclude one of the modules. This problem can be caused if servers are included in the OpenAPI document and the server URL does not end with the context root of the application (for example, because the application is accessed at a different URL through a reverse proxy).
+
+OPENAPI_MERGE_PROBLEM_EXTENSION_CLASH=The {0} extension key from the {1} module clashes with the same extension key from the {2} module. {1} cannot be merged.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.openapi.internal.common/resources/io/openliberty/microprofile/openapi/internal/resources/OpenAPI.nlsprops]: {1} cannot be merged.
-->
The {1} module cannot be merged.

**Review by dmuelle — CHANGES_REQUESTED** [review]

**Comment by Azquelt** [comment]
#build
#libby

**Review by dmuelle — APPROVED** [review]
LGTM

**Pull request merged**